device/allowedips.go

```diff
@@ -223,10 +223,23 @@ func (table *AllowedIPs) EntriesForPeer(peer *Peer, cb func(prefix netip.Prefix)
 	}
 }
 
+func (table *AllowedIPs) SetPeerPrefixes(peer *Peer, prefixes []netip.Prefix) {
```

Comment:
We're going to need to hold this with care, and it's almost tempting to suggest fixing it wireguard-go side, though it's also sort of a pre-existing issue:

If Peer A and Peer B have the same routes, then the last write wins.

If Peer B is removed, the route becomes un-routed.

A typical user expectation would more likely be that Peer A becomes the destination instead.

I believe we do some guarding of this on our side at quite a distance away from here, but it feels risky.

Reply:
oh, I think I can unexport this now. I think this was from when I was waffling on the API I wanted. But this can be an internal detail now.